File: C++/converter.cpp
#include <iostream>
using namespace std;

class Unit {

    public:
        int gram(string unit, int value) {
            
            if ((unit.compare("ounce") == 0)) {
                return (value * 0.03527396);    
            } else if ((unit.compare("lbs") == 0)) {
                return (value * 0.002204623);
            } else if ((unit.compare("kg") == 0)) {
                return (value * 0.001);
            } else {return 0;}
        }

        int meter(string unit, int value) {

            if ((unit.compare("in") == 0)) {
                return (value * 39.37008);    
            } else if ((unit.compare("ft") == 0)) {
                return (value * 3.28084);
            } else if ((unit.compare("cm") == 0)) {
                return (value * 100);
            }  else if ((unit.compare("mm") == 0)) {
                return (value * 1000);
            } else {return 0;}
    }
};

int main() {

    int value;
    string unitBeingConverted, unitConvertedTo;
    Unit converter;

    cout << "What is your starting value?: ";
    cin >> value; 
    cin >> unitBeingConverted;
    cout << endl;

    cout << "What do you want to convert to?: \n";
    cin >> unitConvertedTo;

    if ((unitBeingConverted.compare("g") == 0)) {
        cout << converter.gram(unitConvertedTo, value) << endl;
    } else if ((unitBeingConverted.compare("m") == 0)) {
        cout << converter.gram(unitConvertedTo, value) << endl;
    }
    
    return 0;
}

File: Java/Converter.java
import java.util.Scanner;

public class Converter {

    public static double gram(int value, String convertingTo) {
        if (convertingTo.equals("oz")) {
            return (value * 0.03527396);
        } else if (convertingTo.equals("lbs")) {
            return (value * 0.002204623);
        } else {
            System.out.println("Error");
            return 0;
        }
    }

    public static double ounce(int value, String convertingTo) {
        if (convertingTo.equals("g")) {
            return (value * 28.34952);
        } else if (convertingTo.equals("lbs")) {
            return (value * 0.0625);
        } else {
            System.out.println("Error");
            return 0;
        }
    }

    public static double meter(int value, String convertingTo) {
        if (convertingTo.equals("ft")) {
            return (value * 3.28084);
        } else if (convertingTo.equals("km")) {
            return (value * 0.001);
        } else {
            System.out.println("Error");
            return 0;
        }
    }

    public static void main(String[] arg) {
        Scanner scanner1 = new Scanner(System.in);

        System.out.println("What is your value?: ");
        String value = scanner1.nextLine();
        System.out.println("What is the unit?: ");
        String convertingFrom = scanner1.nextLine();
        System.out.println("What do you want to convert to?: ");
        String convertingTo = scanner1.nextLine();

        scanner1.close();

        int parsed_value = Integer.parseInt(value);

        if (convertingFrom.equals("g")) {
            System.out.println(gram(parsed_value, convertingTo));
        } else if (convertingFrom.equals("oz")) {
            System.out.println(ounce(parsed_value, convertingTo));
        } else if (convertingFrom.equals("m")) {
            System.out.println(meter(parsed_value, convertingTo));
        } else {
            System.out.println("Error");
        }
    }
}
